### PR TITLE
route selection based on user input

### PR DESCRIPTION
**Pull request by pawan-tezsure**

### CODE REVIEW

File: src/Components/SwapDetails.js [hunk 1]
```diff
@@ -6,14 +6,12 @@ import { tokens } from '../constants/swapPage';
 
 const SwapDetails = (props) => {
   const swapRoute = useMemo(() => {
-    if (props.routeData.bestRoute?.path.length > 2) {
-      return props.routeData.bestRoute.path.map((tokenName) =>
-        tokens.find((token) => token.name === tokenName),
-      );
+    if (props.routePath?.length > 2) {
+      return props.routePath.map((tokenName) => tokens.find((token) => token.name === tokenName));
```
Comment (vijaypushkin) [inline src/Components/SwapDetails.js:10]: format with prettier

File: src/Components/SwapDetails.js [hunk 2]
```diff
@@ -149,6 +147,7 @@ SwapDetails.propTypes = {
   // midTokens: PropTypes.any,
   tokenIn: PropTypes.any,
   tokenOut: PropTypes.any,
+  routePath: PropTypes.any,
```
Comment (vijaypushkin) [inline src/Components/SwapDetails.js:150]: remove `routeData` from propTypes

File: src/Components/SwapTabsContent/SwapTab.js [hunk 3]
```diff
@@ -2,23 +2,20 @@ import PropTypes from 'prop-types';
...
 import Button from '../Ui/Buttons/Button';
 import {
-  computeTokenOutForRouteBaseByOutAmountV2,
-  computeTokenOutForRouteBaseV2,
-  swapTokenUsingRouteV2,
-} from '../../apis/swap/swap-v2';
+  computeTokenOutForRouteBaseV3,
+  swapTokenUsingRouteV3,
+  computeTokenOutForRouteBaseByOutAmountV3,
+} from '../../apis/swap/swap-v3';
```
Comment (vijaypushkin) [inline src/Components/SwapTabsContent/SwapTab.js:13]: keep it as `v2`

File: src/Components/SwapTabsContent/SwapTab.js [hunk 4]
```diff
@@ -36,72 +33,53 @@ const SwapTab = (props) => {
...
+            totalFees: res.bestRoute.computations.fees[res.bestRoute.computations.fees.length - 1],
+            minimumOut: res.bestRoute.computations.minimumOut,
+            finalMinimumOut:
+              res.bestRoute.computations.minimumOut[
+                res.bestRoute.computations.minimumOut.length - 1
+              ],
+            priceImpact: res.bestRoute.computations.priceImpact,
+          },
+        });
+        console.log(res);
```
Comment (vijaypushkin) [inline src/Components/SwapTabsContent/SwapTab.js:80]: remove console.log

File: src/apis/swap/swap-v3.js [hunk 5]
```diff
@@ -0,0 +1,491 @@
+import CONFIG from '../../config/config';
```
Comment (vijaypushkin) [inline src/apis/swap/swap-v3.js:1]: Replace v2 file with this one, keep it as v2

File: src/apis/swap/swap-v3.js [hunk 6]
```diff
@@ -0,0 +1,491 @@
...
+      tokenIn_supply,
+      tokenOut,
+      tokenOut_supply,
+      exchangeFee,
+      tokenOutPerTokenIn,
+      lpTokenSupply,
+      lpToken,
+    };
+  } catch (error) {
+    console.log({ message: 'swap data error', error });
```
Comment (vijaypushkin) [inline src/apis/swap/swap-v3.js:55]: console.error

File: src/apis/swap/swap-v3.js [hunk 7]
```diff
@@ -0,0 +1,491 @@
...
+        bestRoute.path = routeDataResponses[i].path;
+      }
+    }
+    return {
+      success: true,
+      bestRouteUntilNoInput: bestRoute,
+      allRoutes: routeDataResponses,
+    };
+  } catch (err) {
+    console.log(err);
```
Comment (vijaypushkin) [inline src/apis/swap/swap-v3.js:149]: use console.error for errors

File: src/apis/swap/swap-v3.js [hunk 8]
```diff
@@ -0,0 +1,491 @@
...
+  } catch (err) {
+    console.error(err);
+    return {
+      success: false,
+      data: initialData,
+    };
+  }
+};
+
+// TOP
```
Comment (vijaypushkin) [inline src/apis/swap/swap-v3.js:257]: Make this comment as JSDOC

**Review by vijaypushkin — CHANGES_REQUESTED** [review]

File: src/Components/SwapDetails.js (hunk 2 above)
Comment (pawan-tezsure) [reply src/Components/SwapDetails.js:150]: It is still been currently used

File: src/Components/SwapTabsContent/SwapTab.js (hunk 3 above)
Comment (pawan-tezsure) [reply src/Components/SwapTabsContent/SwapTab.js:13]: v3 apis are built on v2 apis, renaming them will break the code.

File: src/Components/SwapDetails.js (hunk 2 above)
Comment (vijaypushkin) [reply src/Components/SwapDetails.js:150]: Okay then

File: src/Components/SwapTabsContent/SwapTab.js (hunk 3 above)
Comment (vijaypushkin) [reply src/Components/SwapTabsContent/SwapTab.js:13]: But we're not using v2 file right?

File: src/Components/SwapTabsContent/SwapTab.js (hunk 4 above)
Comment (pawan-tezsure) [reply src/Components/SwapTabsContent/SwapTab.js:80]: Done

File: src/apis/swap/swap-v3.js (hunk 6 above)
Comment (pawan-tezsure) [reply src/apis/swap/swap-v3.js:55]: Done

File: src/apis/swap/swap-v3.js (hunk 7 above)
Comment (pawan-tezsure) [reply src/apis/swap/swap-v3.js:149]: Done

**Review by vijaypushkin — APPROVED** [review]

**Pull request merged**